src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,122 @@
+package nextstep.subway.domain;
+
+import com.google.common.collect.Lists;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import nextstep.subway.dto.SectionResponse;
+import nextstep.subway.value.ErrMsg;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(mappedBy = "line", cascade = CascadeType.ALL)
+    private final List<Section> sections = Lists.newArrayList();
+
+    public void add(Section section) {
+        if(sections.isEmpty()){
+            sections.add(section);
+            return;
+        }
+        validateSection(section);
+        if(!addFirstOrLastSection(section)){
+            addSectionInMiddle(section);
+        }
+    }
+
+    private void validateSection(Section section) {
+        List<Station> stations = getAllStations();
+        // 둘다 포함된 경우 에러
+        if (stations.contains(section.getUpStation()) && stations.contains(section.getDownStation())){
```

Comment:
```suggestion
        if (isContain(section)){
```

조건코드를 설명 할 수 있는 표현이 있을것 같아요
그러한 측면에서 의미있는 이름을 가진 메소드로 분리하면 어떨까요?
가독성 측면이나 한가지 책임을 담당하는 메소드를 분리함으로써 유지보수성도 좋아질것 같구요 :thinking: